sycl/source/detail/config.hpp

```diff
@@ -579,6 +579,34 @@ template <> class SYCLConfig<SYCL_REDUCTION_PREFERRED_WORKGROUP_SIZE> {
   }
 };
 
+template <> class SYCLConfig<SYCL_ENABLE_FUSION_CACHING> {
+  using BaseT = SYCLConfigBase<SYCL_ENABLE_FUSION_CACHING>;
+
+public:
+  static bool get() {
+    constexpr bool DefaultValue = true;
+
+    const char *ValStr = getCachedValue();
+
+    if (!ValStr)
+      return DefaultValue;
+
+    return std::stoi(ValStr);
```

Comment:
It should also be specified in the aforementioned documentation of the option, but what is the expected behavior if I pass some non-integer value? `std::stoi` will throw, so if we want it to fall back to the default value in that case we would probably want to catch it.

Reply:
Agreed, error is caught now.